### ISSUE TITLE
cmd/go: `go work use $DIR` adds different paths from `go work init $DIR`

### ISSUE THREAD

**Issue by bcmills** [golang/go#51449]
Both `go work use` and `go work init` can be used to add paths to a `go.work` file. However, the paths that they add differ in whether they include a `./` prefix.

We should figure out which form they should canonically use and update them to match.

```
~/src/golang.org/x/exp$ gotip version
go version devel go1.19-6da16b6ad Tue Mar 1 23:49:01 2022 +0000 linux/amd64

~/src/golang.org/x/exp$ gotip work init shiny/

~/src/golang.org/x/exp$ cat go.work
go 1.19

use ./shiny

~/src/golang.org/x/exp$ rm go.work

~/src/golang.org/x/exp$ gotip work init

~/src/golang.org/x/exp$ go work use shiny/

~/src/golang.org/x/exp$ cat go.work
go 1.19

use shiny
```

CC @matloob

**Comment by matloob** [comment]
This is fixed in https://go-review.googlesource.com/c/go/+/389854, which is being cherry-picked onto the release branch in https://go-review.googlesource.com/c/go/+/389298.

**Comment by dmitshur** [comment]
@matloob In that case, can this release-blocking issue be closed in favor of #51448 (also a release blocking issue, the one that will be marked closed when CL 389298 lands on release-branch.g1.18)?

**Comment by bcmills** [comment]
Yeah, I kind of smooshed the two fixes into one CL and forgot to refer to both issues. Let's go ahead and close this one, since it's fixed at head and in the backport stack.
